runtime/java.go

```diff
@@ -83,50 +75,44 @@ main: <name>
 #                   - /
 #                   - /package1
 classpath:
-   <list>
+   - <list>
+
+# OPTIONAL
+# Initial and maximum JVM memory size.
+# Example value: xms: 512m
+xms: <value>
+xmx: <value>
+
+# OPTIONAL
+# A list of JVM args.
+# Example value: jvm_args:
+#                   - -Djava.net.preferIPv4Stack=true
+#                   - -Dhadoop.log.dir=/hdfs/logs
+jvm_args:
+   - <list>
 
 # OPTIONAL
 # A list of command line args used by the application.
 # Example value: args:
 #                   - argument1
 #                   - argument2
 args:
-   <list>
-
-# OPTIONAL
-# A list of JVM args (e.g. Xmx, Xms)
-# Example value: jvmargs:
-#                   - Xmx1000m
-#                   - Djava.net.preferIPv4Stack=true
-#                   - Dhadoop.log.dir=/hdfs/logs
-jvmargs:
-   <list>
+   - <list>
 ` + conf.CommonRuntime.GetYamlTemplate()
 }
 
 //
 // Utility
 //
 
-func (conf javaRuntime) GetCommandLine() string {
-	var cp, args string
-
-	if len(conf.Classpath) > 0 {
-		cp = "-cp " + strings.Join(conf.Classpath, ":")
-	}
-
-	if len(conf.Args) > 0 {
-		args = strings.Join(conf.Args, " ")
-	}
+func (conf javaRuntime) concatJvmArgs() string {
+	res := strings.Join(conf.JvmArgs, " ")
 
-	return strings.TrimSpace(fmt.Sprintf("%s %s %s", cp, conf.Main, args))
-}
-func (conf javaRuntime) GetJvmArgs() string {
-	vmargs := ""
-
-	for _, arg := range conf.JvmArgs {
-		vmargs += fmt.Sprintf("-%s ", arg)
+	// This is a workaround since runscript is currently unable to
+	// handle empty environment variable as a parameter. So we set
+	// dummy value unless user provided some actual value.
+	if res == "" {
+		return "-Dx=y"
```

Comment:
Would a form like the one below be more "readable"?

```
if len(conf.JvmArgs) > 0 {
    return strings.Join(...)
} else {
    return "-Dx=y"
}
```

I also thought whether just setting `res = "-Dx=y"` and let it return in L117.